### ISSUE TITLE
Project Gutenberg

### ISSUE THREAD

**Issue by davidar** [ipfs/archives#29]
The first thing I mirrored to IPFS was [a small subset of Project Gutenberg](http://ipfs.io/ipfs/QmPn2Ezc1t8juabugt5BYjebrqcqbJvdgUt4kFKXQBqVJD/), so I'm definitely interested in getting the whole thing into IPFS, as both @rht (#14) and @simonv3 (https://github.com/simonv3/ipfs-gutenberg) have suggested.

Making an issue to coordinate this.

**Comment by rht** [comment]
This is just an `rsync` away, really.
Currently running it on pollux.

**Comment by davidar** [comment]
@rht is there enough free disk space on Pollux?

**Comment by rht** [comment]
(didn't check)

**Comment by rht** [comment]
https://www.gutenberg.org/wiki/Gutenberg:Mirroring_How-To says it is at least 650 GB (could have been doubled).
Pollux has 13 GB left.

But anyway, the mirroring is a one-liner.

**Comment by simonv3** [comment]
@rht Yeah, what makes this difficult is the amount of disk space - I don't think many people have that amount of space lying around for this.

It's been suggested by some people to shard the collection and just make sure people hosting those bits keep their things in sync independently. There's also been talk about this tool: https://github.com/ipfs/notes/issues/58

**Comment by simonv3** [comment]
We could also just pitch in x amount for an Amazon instance (or some other host) of that amount, and just pay that that?

**Comment by rht** [comment]
Hmm, rsync doesn't have seek so at least the first 'download -> hash' needs the TB storage to contain it.

Either
1. https://aws.amazon.com/s3/reduced-redundancy/ ~$24/month.
2. http://www.amazon.com/Green-1TB-Desktop-Hard-Drive/dp/B006GDVREI ~$50 (can be repurposed for other archivals, once the PG hash has been sharded).

For now, to do partial backup, `ipfs object get` can be used for each of the nodelinks that form parts of the root hash.

**Comment by rht** [comment]
(and both storage came from amazon)

**Comment by rht** [comment]
`ipfs check-redundancy $hash` would be useful.

**Comment by davidar** [comment]
@jbenet @lgierth SEND MORE DISKS...

![](http://25.media.tumblr.com/tumblr_m8050kZcnb1qzcgluo1_1280.png)

Also see ipfs/infrastructure#89

**Comment by davidar** [comment]
> ipfs check-redundancy $hash would be useful.

@rht Yeah, what I really want to do is have a "click to pin" button on the archive homepage, people select how much storage they want to donate, and the tool randomly selects an appropriate subset of the least-redundant blocks and pins them to the local daemon.

CC: @whyrusleeping

**Comment by whyrusleeping** [comment]
that would be cool. could have our service enumerate providers for each block under a given archive root, then assign blocks with the least number of providers to the next person who requests.

**Comment by rht** [comment]
Should be normalized based on the blocks demand curve.

**Comment by jbenet** [comment]
https://github.com/ipfs/notes/issues/58

**Comment by jbenet** [comment]
We can get more storage nodes, if necessary
